glue/plugins/join_on_key/link_helpers.py

```diff
@@ -0,0 +1,52 @@
+from glue.config import link_helper
+from glue.core.link_helpers import LinkCollection
+
+
+@link_helper(category="Join")
+class Join_Link(LinkCollection):
+    cid_independent = False
+
+    display = "Join on ID"
+    description = "Join two datasets on a common indetifier or index. \
+This is similar to a database join in that subsets defined \
+on one dataset can always propogate through joins."
```

Comment:
I would perhaps give a little more detail and an example - to show how it differs to a classic link?

Reply:
Taking inspiration from [the docs](http://docs.glueviz.org/en/stable/gui_guide/merging.html) I now describe this as:

> Join two datasets on a common ID. Other links 
> in glue connect data columns (two datasets have 'age' columns but 
> the rows are different objects), while Join on ID connects the same 
> rows/items across two datasets.

There isn't a great deal of space in the UI for a longer description, and concise is better anyway. Do you think this is sufficient?

Reply:
@astrofrog -- if you approve of this new description I think this is good-to-go?